lib/manageiq/appliance_console/message_configuration.rb

```diff
@@ -0,0 +1,134 @@
+
+      BASE_DIR                          = "/opt/kafka".freeze
+      LOGS_DIR                          = "#{BASE_DIR}/logs".freeze
+      CONFIG_DIR                        = "#{BASE_DIR}/config".freeze
+      SAMPLE_CONFIG_DIR                 = "#{BASE_DIR}/config-sample".freeze
+      MIQ_CONFIG_DIR                    = "/var/www/miq/vmdb/config".freeze
+
+      def initialize(options = {})
+        @username, @password = nil
+
+        @username = options[:username] || "admin"
+        @password = options[:password]
+
+        @miq_config_dir_path               = Pathname.new(MIQ_CONFIG_DIR)
+        @config_dir_path                   = Pathname.new(CONFIG_DIR)
+        @sample_config_dir_path            = Pathname.new(SAMPLE_CONFIG_DIR)
+
+        @tools_log4_properties_path        = @config_dir_path.join("", "tools-log4j.properties")
+        @tools_log4_properties_sample_path = @sample_config_dir_path.join("", "tools-log4j.properties")
+        @client_properties_path            = @config_dir_path.join("", "client.properties")
+
+        @keystore_dir_path                 = @config_dir_path.join("", "keystore")
+        @truststore_path                   = @keystore_dir_path.join("", "truststore.jks")
+        @keystore_path                     = @keystore_dir_path.join("", "keystore.jks")
+
+        @messaging_yaml_sample_path        = @miq_config_dir_path.join("", "messaging.kafka.yml")
+        @messaging_yaml_path               = @miq_config_dir_path.join("", "messaging.yml")
+      end
+
+      def create_tools_log_config
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        FileUtils.cp(@tools_log4_properties_sample_path, @tools_log4_properties_path) unless file_found?(@tools_log4_properties_path)
+      end
+
+      def create_client_properties
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        content = <<~CLIENT_PROPERTIES
+          ssl.truststore.location=#{@truststore_path}
+          ssl.truststore.password=#{@password}
+
+          sasl.mechanism=PLAIN
+          security.protocol=SASL_SSL
+          sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \\
+            username=#{@username} \\
+            password=#{@password} ;
+        CLIENT_PROPERTIES
+
+        File.write(@client_properties_path, content) unless file_found?(@client_properties_path)
+      end
+
+      def configure_messaging_yaml
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        return true if file_found?(@messaging_yaml_path)
+
+        messaging_yaml = YAML.load_file(@messaging_yaml_sample_path)
+
+        messaging_yaml["production"]["hostname"] = @server_hostname
+        messaging_yaml["production"]["port"] = 9093
+        messaging_yaml["production"]["username"] = @username
+        messaging_yaml["production"]["password"] = @password
+
+        File.write(@messaging_yaml_path, messaging_yaml.to_yaml) unless file_found?(@messaging_yaml_path)
```

Comment:
The `unless` shouldn't be needed since we checked a few lines above.

Reply:
@bdunne Good catch! Fixed by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/130/commits/5dcf698752f575400a690b9056d135ea22e7ec3b